src/components/network/layers/arrow-layer.js

```diff
@@ -0,0 +1,342 @@
+                let prevPosition;
+                positions.forEach((position, index) => {
+                    // compute distance with previous position
+                    let segmentDistance = 0;
+                    if (prevPosition) {
+                        ruler = cheapRuler(position[1], 'meters');
+                        segmentDistance = ruler.distance([prevPosition[1], prevPosition[0]],
+                                                         [position[1], position[0]]);
+                    }
+                    prevPosition = position;
+
+                    // fill line positions texture
+                    linePositionsTextureData.push(position[0]);
+                    linePositionsTextureData.push(position[1]);
+
+                    linePointCount++;
+
+                    // fill line distance texture
+                    lineDistance += segmentDistance;
+                    lineDistancesTextureData.push(lineDistance);
+                });
+            }
+            lineAttributes.set(line, {
+                distance: lineDistance,
+                positionsTextureOffset: linePositionsTextureOffset,
+                distancesTextureOffset: lineDistancesTextureOffset,
+                pointCount: linePointCount
+            });
+        });
+
+        const stop = performance.now()
+        console.info(`Texture data created in ${stop -start} ms`);
+
+        return {
+            linePositionsTextureData,
+            lineDistancesTextureData,
+            lineAttributes
+        }
+    }
+
+    updateState({props, oldProps, changeFlags}) {
+        super.updateState({props, oldProps, changeFlags});
+        if (changeFlags.extensionsChanged) {
+            const {gl} = this.context;
+
+            const {
+                model
+            } = this.state;
+            if (model) {
+                model.delete();
+            }
+
+            const {
+                linePositionsTextureData,
+                lineDistancesTextureData,
+                lineAttributes
+            } = this.createTexturesStructure(props);
+
+            const linePositionsTexture = this.createTexture2D(gl, linePositionsTextureData, 2, GL.RG32F);
+            const lineDistancesTexture = this.createTexture2D(gl, lineDistancesTextureData, 1, GL.R32F);
+
+            this.setState({
+                model: this._getModel(gl),
+                linePositionsTexture: linePositionsTexture,
+                lineDistancesTexture: lineDistancesTexture,
+                lineAttributes: lineAttributes,
+                timestamp: 0,
+                stop: false
+            });
+
+            this.getAttributeManager().invalidateAll();
+
+            if (this.props.animated) {
+                this.startAnimation();
+            }
+        }
+    }
+
+    animate(timestamp) {
+        if (this.state.stop) {
+            return;
+        }
+        this.setState({
+            timestamp: timestamp,
+        });
+        this.startAnimation();
```

Comment:
since this is running at 60 fps, maybe it's best to avoid recreating the lambda everytime and juste call requestAnimationFrame(animate) ?